_overviews/scala3-macros/best-practices.md

```diff
@@ -55,7 +55,101 @@ val y: Expr[Int] = ...
 '{ (x: Int) => ${ body('x, y) } }
 ```
 
+## Quotes Reflect
+
+For this section, consider the following setup:
+
+```scala
+object Box:
+  sealed trait Base
+  case class Leaf(x: Int) extends Base
+
+// Quotes in contextual scope
+val boxTpe = TypeRepr.of[Box.type]
+val baseTpe = TypeRepr.of[Box.Base]
+val baseSym = baseTpe.typeSymbol
+val leafTpe = TypeRepr.of[Box.Leaf]
+val leafSym = leafTpe.typeSymbol
+```
+
+### Avoid `Symbol.tree`
+
+`Symbol.tree` returns the `Tree` associated to the symbol. Be careful when using this
+method as the tree for a symbol might not be defined. When the code associated to the symbol
+is defined in a different moment than this access, if the `-Yretain-trees` compilation option
+is not used, then the `tree` of the symbol will not be available. Symbols originated from
+Java code do not have an associated `tree`.
+
+### Obtaining a `TypeRepr` from a `Symbol`
+
+In the previous paragraph we saw that `Symbol.tree` should be avoided and therefore
+you should not use `Symbol.tree.tpe`.
+Thus to obtain the `TypeRepr` corresponding to a `Symbol`, it is recommended
+to use `TypeRepr.memberType`
+
+We can obtain the `TypeRepr` of `Leaf` in two ways:
+  1. `TypeRepr.of[Box.Leaf]`
+  2 `boxTpe.memberType(leafSym)`, in other words we request
+  the `TypeRepr` of the member of `Box` whose symbol is equal to the symbol of sym
+
+### Use `Symbol`s to compare definitions
+
+Read more about Symbols [here][symbol].
+
+Symbols allow comparing definitions using `==`:
+```scala
+leafSym == baseSym.children.head // Is true
+```
+
+However, `==` on `TypeRepr`s does not produce the same result:
+```scala
+boxTpe.memberType(baseSym.children.head) == leafTpe // Is false
+```
+
+### Obtaining a Symbol for a type
+
+Because `Symbol.tree` should be avoided (read more above),
+you should not use `TypeTree.of[T].tpe.typeSymbol` to get the symbol of
```

Comment:
This one does not use `Symbol.tree`. `TypeRepr.of[T]` does the equivalent of `TypeTree.of[T].tpe` internally.

I would say this is only to make it shorter.